x-pack/docs/en/security/securing-communications/security-basic-setup.asciidoc

```diff
@@ -0,0 +1,277 @@
+ensure that communication between the nodes is encrypted.
+
+Now that you've generated a certificate authority and certificates, you'll
+update your cluster to use these files.
+
+NOTE: Complete the following steps for each node in your cluster. To join the
+same cluster, all nodes must share the same `cluster.name` value.
+
+.  Open the `ES_PATH_CONF/elasticsearch.yml` file and make the following
+changes:
+
+   a. Add the `cluster-name` setting and enter a name for your cluster:
++
+[source,yaml]
+----
+cluster.name: my-cluster
+----
+
+b. Add the `node.name` setting and enter the name of the certificate that
+you generated for this node. For simplicity, it's good practice for this value
+to match the certificate name that you defined in your `certificates.yaml` file:
++
+[source,yaml]
+----
+node.name: node-1
+----
+
+c. Add the following settings to enable internode communication and provide
+access to the node's certificate:
++
+[source,yaml]
+----
+xpack.security.transport.ssl.enabled: true
+xpack.security.transport.ssl.verification_mode: certificate
+xpack.security.transport.ssl.client_authentication: required
+xpack.security.transport.ssl.keystore.path: <node-name>.p12
+xpack.security.transport.ssl.truststore.path: <node-name>.p12
+----
+
+.  If you entered a password when creating the node certificate, run the following commands to store the password in the {es} keystore:
++
+--
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.transport.ssl.keystore.secure_password
+----
+
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.transport.ssl.truststore.secure_password
+----
+--
+
+.  Complete the previous steps for each node in your cluster.
+
+.  Restart {es}. The method for <<starting-elasticsearch,starting>> and <<starting-elasticsearch,stopping>> {es} varies depending on how you installed it.
++
+For example, if you installed {es} with an archive distribution
+(`tar.gz` or `.zip`), you can enter `Ctrl+C` on the command line to stop
+{es}.
++
+WARNING: You must perform a full cluster restart. Nodes that are configured to
+use TLS for transport cannot communicate with nodes that use unencrypted transport connection (and vice-versa).
+
+*Next*: <<add-built-in-users,Add the built-in users to {kib}>>
+
+[[add-built-in-users]]
+==== Add the built-in users to {kib}
+
+When the {es} security features are enabled, users must log in to
+{kib} with a valid username and password.
+
+{kib} also performs some background tasks that require use of the built-in
+`kibana_system` user.
+
+You'll configure {kib} to use the built-in `kibana_system` user and the
+password that you created earlier.
+
+1. From the directory where you installed {kib}, run the following commands
+   to create the {kib} keystore and add the secure settings:
+
+   a. Create the {kib} keystore:
++
+[source,shell]
+----
+./bin/kibana-keystore create
+----
+
+   b. Add the {es} user to the {kib} keystore:
```

Comment:
Do we want to encourage people to add the **username** to the keystore? I would expect the username to be in clear text in the yaml, and have only the password in the  keystore.

Reply:
Thanks for bringing this up @tvernum. I'm going to change these instructions so that users add `elasticsearch.username` to `kibana.yml`, and then add the `elasticsearch.password` to the Kibana keystore. I'm also changing these instructions to the `elastic` user. Using `kibana_system` doesn't seem to work, and makes the instructions more confusing.